ls1intum/Artemis
# `Development`: Replace learning goal with competency in server file and class names

### Checklist
#### General
#### Server

### Motivation and Context
This is a part of follow-up PRs on #6388 There's currently an inconsistency between the usage of 'learning goal' and 'competency' throughout the system. This PR only addresses the **server file and class names**.

### Description
Replaced all occurrences of the term 'learning goal' with 'competency' on server file and class names. Also, some fields and functions are renamed to match the new class names. 

**Note**: there can be some fields and annotations left unchanged as they affect the database structure. This is okay since this PR doesn't aim to make any database-related changes.

**Note 2**: I had to rename some fields in client fields as well to match the models sent from the server.

### Steps for Testing
Prerequisites:
- 1 Course with competencies
- 1 Instructor

1. Log in to Artemis as Instructor
2. Navigate to Course Administration
3. Go to the Competency Management page of the course
4. Test CRUD operations (+import) for competencies
5. Link this competency to a Lecture Unity or Exercise and see if completing it updates the competency progress for the user

### Review Progress
#### Code Review
#### Manual Tests
